collect.py

```diff
@@ -89,21 +90,20 @@ def do_something(driver, elem_id=None):
         random.shuffle(children)
 
         for child in children:
-            elem_id = child.get_attribute('id')
 
-            # We need to store the ID in order to replicate what we are doing, so we
-            # have to skip elements with no ID.
-            if elem_id == '':
-                continue
+            # Get all the attributes of the child.
+            child_attributes = driver.execute_script('var elem_attribute = {}; for (i = 0; i < arguments[0].attributes.length; i++) { elem_attribute[arguments[0].attributes[i].name] = arguments[0].attributes[i].value }; return elem_attribute;', child)
 
             # If the element is not displayed or is disabled, the user can't interact with it. Skip
             # non-displayed/disabled elements, since we're trying to mimic a real user.
             if not child.is_displayed() or not child.is_enabled():
                 continue
 
-            elem = child
-            break
+            if elem_attributes is None or elem_attributes == child_attributes:
+                elem = child
+                break
     else:
+        elem_id = elem_attributes['id']
         elem = driver.find_element_by_id(elem_id)
```

Comment:
This is still only using the ID. It means we will find many more possible elements in the first browser, but then won't be able to reproduce the same steps in the second browser.

Reply:
@marco-c We are able to reproduce same steps in second browser as in line 80 I do check if the ```id``` is present in ```elem_attributes```. In most cases it is not found so the code enters the ```if``` condition and finds the element on the basis of attributes. Images in #15 demonstrate how the code reproduces the same steps for the second browser.

Reply:
Oh you're right, sorry I hadn't noticed.
I would refactor this a bit so that the first branch is taken when we are looking for an element to choose, and the second branch when we already know the element we want to use (either with the attributes or the ID).

Reply:
@marco-c  I didn't understand what do you mean by refactoring it? Should I create two functions?

Reply:
I mean something like this:
```Python
if elem.attributes is None:
    CODE THAT CHOOSES AN ELEMENT FROM THE PAGE
else:
    FIND AN ALREADY CHOSEN ELEMENT BY ID OR USING ALL THE ATTRIBUTES
```

So the first branch implements the "choose random element" part, the second branch implements the "I already know what element I want to use, let's find it on the page" part.

Reply:
@marco-c  I made the required change. Please check.